app/javascript/mastodon/components/status_content.js

```diff
@@ -46,10 +51,15 @@ export default class StatusContent extends React.PureComponent {
     }
   }
 
+  componentDidMount () {
+    this._updateStatusLinks();
+  }
+
   componentDidUpdate () {
     if (this.props.onHeightUpdate) {
       this.props.onHeightUpdate();
     }
+    this._updateStatusLinks();
```

Comment:
That fails, however, for the CW mention placeholder, which also exhibit the same issues; opening and closing the CW causes the placeholder mention links to lose their handlers, and obviously the status hasn't changed so this would miss it.